### ISSUE TITLE
[feature] have ids be recordId(td:string, id: string) instead of string

### ISSUE THREAD

**Issue by djkato** [horvbalint/surreal-ts#12]
Using this and loving it, super useful. Thank you for making it!
One of my biggest annoyances right now is that `id` field on all Out and In types are strings, and not RecordIds. I don't expect complex ID types to work like arrays and such, but at least separate table and id into two, so I don't get these everywhere:

![Image](https://github.com/user-attachments/assets/6dd132d3-a4b8-4afc-847e-29183bf56987)

Or maybe I'm using the types wrong?
Thanks!

**Comment by horvbalint** [comment]
Nope, you are absolutely right. The main reason I abandoned this project last time was exactly this.
I used Surreal as a DB only so when I sent the data from my backend to the frontend it got serialized to json and back, so for me they arrived as strings. 
Later I found out that for someone using the surreal js sdk some types (ex.: Id or Duration) are not correct and I felt no energy in me to continue.
But this can be solved quite easily, I will introduce a new config flag '--for-sdk' or similar and if this is set than the correct types will be used. I think this flag should also default to true as I suspect most people are using Surreal with the official SDK. I will try to find some time to work on this, but if you feel like it a PR is also welcomed.

I am happy you find this package useful, even in its current half-baked form :)

**Comment by horvbalint** [comment]
Can you show/tell me what jsonify looks like or where it comes from?

I think this is (more than) OK. What you see there is a Typescript's [Template Literal Types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) which restrict what strings are accepted. So in this case it show that id is a string that starts with `app_category:` that is followed by an other string part which is really cool.
I could not replicate this, but I wish I could :D Simple calling `.toJson()` on an id field does not produce this for me, I think `jsonify` is something special.

**Comment by horvbalint** [comment]
Thanks, I have to take a look at the js sdk again, it changed a lot since I last saw it.

**Comment by djkato** [comment]
Hey, I got another feedback, but I'm not sure if it could be actionable:
When I use svelte to construct a surrealdb type client-side before upserting it to db, I found that Sveltes reactive system doesn't work with RecordId well. I can't bind OutThing.id.id, binding OutThing.id would show the table part too. Thus I found myself using 
```ts
const smt = get_things<Omit<OutThing, "id"> & {id:string}>()
```
And later turning it into the proper RecordId because of this. I'm not sure still why there's OutThing and InThing, but would it make sense for one of them to just be a string? Or maybe a cool exported helper function that helps turning RecordId to just the string id and back for these kinds of usecases.

I don't know how to solve this, just giving what I find as an annoyance. 

Cheers :)

**Comment by horvbalint** [comment]
@djkato Could you open a new issue ticket for this and explain the issue a bit more in-depth for me? I have not used Svelte yet and could not follow your line of thought. Thanks :)
